lm/srilm.py

```diff
@@ -0,0 +1,553 @@
+__all__ = [
+    "CountNgramsJob",
+    "ComputeNgramLmJob",
+    "ComputeNgramLmPerplexityJob",
+    "ComputeBestMixJob",
+    "InterpolateNgramLmJob",
+    "PruneLMWithHelperLMJob",
+]
+
+import os
+import shutil
+
+from enum import Enum
+from typing import Dict, List, Optional
+
+from sisyphus import tk, Job, Task
+
+from i6_core.util import create_executable, relink
+
+
+class CountNgramsJob(Job):
+    """
+    Count ngrams with SRILM
+    """
+
+    def __init__(
+        self,
+        ngram_order: int,
+        data: tk.Path,
+        count_exe: tk.Path,
+        *,
+        extra_count_args: Optional[List[str]] = None,
+        mem_rqmt: int = 48,
+        time_rqmt: float = 24,
+        cpu_rqmt: int = 1,
+        fs_rqmt: str = "100G",
+    ):
+        """
+        :param ngram_order: Maximum n gram order
+        :param data: Input data to be read as textfile
+        :param extra_count_args: Extra arguments for the execution call e.g. ['-unk']
+        :param count_exe: Path to srilm ngram-count executable
+        :param mem_rqmt: Memory requirements of Job (not hashed)
+        :param time_rqmt: Time requirements of Job (not hashed)
+        :param cpu_rqmt: Amount of Cpus required for Job (not hashed)
+        :param fs_rqmt: Space on fileserver required for Job, example: "200G" (not hashed)
+
+        Example options/parameters for count_args:
+        -unk
+        """
+        self.ngram_order = ngram_order
+        self.data = data
+        self.count_args = extra_count_args if extra_count_args is not None else ["-unk"]
+        self.count_exe = count_exe
+
+        self.out_counts = self.output_path("counts", cached=True)
```

Comment:
i am more a fan of deciding caching on the input side, but as this anyway inconsisten everywhere lets keep it